sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/ReplaceNullWithFalseInPredicate.scala

```diff
@@ -53,6 +53,7 @@ object ReplaceNullWithFalseInPredicate extends Rule[LogicalPlan] {
   def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     case f @ Filter(cond, _) => f.copy(condition = replaceNullWithFalse(cond))
     case j @ Join(_, _, _, Some(cond), _) => j.copy(condition = Some(replaceNullWithFalse(cond)))
+    case d @ DeleteFromTable(_, Some(cond)) => d.copy(condition = Some(replaceNullWithFalse(cond)))
```

Comment:
how about UPDATE and MERGE? they have conditions as well.

Reply:
It's beyond of this scope of this PR~

Reply:
Yeah, it's fine to fix it separately. But still how about fixing them in a separate PR @aokolnychyi?

Reply:
Sure, I guess so. There are many things on the way in this area.

Reply:
I will follow up for sure! Just wanted to keep the scope small.

Reply:
Thanks, @aokolnychyi.